packages/pubsub/src/connection-pool.js

```diff
@@ -177,6 +190,23 @@ ConnectionPool.prototype.createConnection = function() {
     var connection = client.streamingPull(self.metadata_);
     var errorImmediateHandle;
 
+    if (self.isPaused) {
+      connection.pause();
+    }
+
+    self.once(CHANNEL_ERROR, function(err) {
+      connection.cancel();
+    });
+
+    self.once(CHANNEL_READY, function() {
+      connection.isConnected = true;
```

Comment:
This seems to be setting a property on a gRPC connection object. Is it supposed to be used this way? (Without knowing more, it seems like it's setting an internal property). I see this was part of the code before, but just wondering.

Reply:
It's not setting an internal property, but I can store it in a different way if you prefer.